llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/SandboxVectorizer.h

```diff
@@ -24,6 +25,8 @@ class SandboxVectorizerPass : public PassInfoMixin<SandboxVectorizerPass> {
   AAResults *AA = nullptr;
   ScalarEvolution *SE = nullptr;
 
+  std::unique_ptr<sandboxir::Context> Ctx;
```

Comment:
Does this need to be a member?

Reply:
I think it does need to be a member, here is why:
If we create the Context within the scope of `runImpl()`, like we used to, then `Ctx` goes out of scope once we return from `runImpl()`, but the problem is that the nested members of the `SandboxVectorizerPass`, like the Scheduler, get destroyed after the scope goes out of scope. This means that if we try to unregister the scheduler's callbacks within the Scheduler destructor we access freed memory.

Not sure how else we could fix this in a clean way. Having a member context seems to be the easiest fix.